### DISCUSSION TITLE
channel gain and users location

### DISCUSSION THREAD

**Discussion by yacine074** [NVlabs/sionna#18] [Q&A] [answered]
I would like to know if it is possible to get the channel gain between a pair of users, or users/BS and specify the users location in this tutorial [sionna tutorial](https://nvlabs.github.io/sionna/examples/Realistic_Multiuser_MIMO_Simulations.html)

**Comment by jhoydis** [comment, marked as answer]
Hi,

The location of users and base stations can be set with the function [set_topolgy](https://nvlabs.github.io/sionna/api/channel.html#sionna.channel.tr38901.UMi.set_topology).
In cell 12 of the tutorial, the model updates the user positions each time it is called through the method ``new_topology``.

In cell 8, the line 

```
h_freq = cir_to_ofdm_channel(frequencies, a, tau, normalize=True)
```
transforms the channel impulse responses into the channel frequency responses which are used for the simulation of OFDM transmissions. This is visualised in cell 9.

You can also call the function [``cir_to_time_channel``](https://nvlabs.github.io/sionna/api/channel.html#sionna.channel.cir_to_time_channel) to obtain the discrete time-domain impulse responses. This process is nicely explained in cells 10-14 of this [tutorial](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html#CDL-Channel-Model).

The 3GPP 38.901 models only define the channels between users and base stations, but not between users.

Also note that pathloss and shadow fading are disabled in the [tutorial](https://nvlabs.github.io/sionna/examples/Realistic_Multiuser_MIMO_Simulations.html) you mention. You could activate both by changing cell 4 to
```
channel_model = UMi(carrier_frequency=carrier_frequency,
                    o2i_model="low",
                    ut_array=ut_array,
                    bs_array=bs_array,
                    direction=direction,
                    enable_pathloss=True,
                    enable_shadow_fading=True)
```
